includes/utilities.php

```diff
@@ -204,3 +243,49 @@ function add_leading_slash( $string ) {
 
 	return '/' . $string;
 }
+
+/**
+ * Update an option with query parameters that should be preserved for
+ * matching redirects.
+ *
+ * @param array $parameters New query string parameters to add to the preserved list.
+ * @return void
+ */
+function add_preserved_query_string_parameters( array $parameters ) {
+	$existing_parameters = get_preserved_query_parameters();
+	$parameters = array_merge( $existing_parameters, $parameters );
+	$parameters = array_unique( $parameters );
+	$parameters = array_filter( $parameters );
+	update_option( PRESERVED_PARAMS_OPTION, implode( ';', $parameters ) );
+}
+
+/**
+ * Return list of query parameters to preserve.
+ *
+ * @return array
+ */
+function get_preserved_query_parameters() : array {
+	return explode( ';', get_option( PRESERVED_PARAMS_OPTION, '' ) );
```

Comment:
Why is this stored as a string, not just an array of strings?

Reply:
Implode / explode is faster than serialize / unserialize and it's a simple array.